tests/test_dto_creation.rs

```diff
@@ -1,136 +1,167 @@
-        let lg_dto_user: LoginDto = user.clone().into();
-        let pf_dto_user: ProfileDto = user.clone().into();
-
-        println!("User to LoginDto = {:?}", lg_dto_user);
-        println!("User to ProfileDto = {:?}", pf_dto_user);
-        assert_eq!(
-            LoginDto {
-                login: user.username.to_string(),
-                password: user.password,
-            },
-            lg_dto_user
-        );
-        //values of user field is being moved only for the test scenario. if used further, use .to_string()
-        assert_eq!(
-            ProfileDto {
-                username: user.username.to_string(),
-                email: user.email.to_string(),
-                firstname: user.firstname,
-                middle_name: None,
-                lastname: user.lastname,
-                age: user.age,
-            },
-            pf_dto_user
-        );
-    }
-
-    #[test]
-    fn test_dto_into_struct() {
-        let person = PersonDto {
-            firstname: "Dessalines".to_string(),
-            lastname: "Jean Jacques".to_string(),
-            email: Some("dessalines@mail.ht".to_string()),
-        };
-
-        let user_from_person: User = person.into();
-
-        assert_eq!("Dessalines", user_from_person.firstname);
-        assert_eq!("Jean Jacques", user_from_person.lastname);
-        assert_eq!("dessalines@mail.ht", user_from_person.email);
-
-        //assert all the remaining fields are all initialized with default value from User::default
-        assert_eq!(User::default().username, user_from_person.username);
-        assert_eq!(User::default().password, user_from_person.password);
-        assert_eq!(User::default().middle_name, user_from_person.middle_name);
-        assert_eq!(User::default().age, user_from_person.age);
-    }
-
-    #[test]
-    fn test_dto_with_builder() {
-        let mut login_dto_builder = LoginDtoBuilder::default();
-        let login_dto = login_dto_builder
-            .login("capois-lamort".into())
-            .password("hello123".into())
-            .build()
-            .expect("Failed to build login dto");
-        println!("LoginDto built with a builder: {:?}", login_dto);
-    }
-
-    #[test]
-    fn test_custom_dto_expression() {
-        let user = User {
-            firstname: "Dessalines".into(),
-            lastname: "Jean Jacques".into(),
-            ..User::default()
-        };
-
-        let custom_dto: CustomDto = user.clone().into();
-
-        assert_eq!(
-            custom_dto.name,
-            format!("{} {}", user.firstname, user.lastname)
-        );
-    }
+  #[mapper(
+    dto="CustomDtoWithAttribute" ,
+    no_builder=true ,
+    map=[ ("email",false) ],
+    derive=(Debug, Clone, Serialize, Deserialize),
+    new_fields=[(
+        "name: String",
+        "concat_str( self.firstname.as_str(), self.lastname.as_str() )",
+        ["#[serde(rename = \"renamed_name\")]"],
+    )],
+    macro_attr=["serde(rename_all = \"UPPERCASE\")"]
```

Comment:
if we can have macro attributes for existing fields as well with the use of map keyword already in use.